components/Alert/AlertsAPI.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { ReactNode, useState, useEffect } from "react"
+import ReactDom from "react-dom"
+import { useTransition } from "react-spring"
+
+import Alert from "./components/Alert/Alert"
+
+import { placementType, AlertProps } from "./components/Alert/AlertInterface"
+import {
+  TopLeftBox,
+  TopRightBox,
+  BottomLeftBox,
+  BottomRightBox,
+  AnimatedDiv,
+  Div,
+} from "./components/StyledAlerts"
+
+export interface AlertsWrapperProps {
+  placement?: placementType
+  getContainer?: () => HTMLElement | ReactNode
+}
+
+export interface AlertsWrapperStates {
+  alerts: Array<AlertProps>
+}
+
+export interface AlertAPIInterface {
+  open: (obj: AlertProps) => void
+  remove?: (obj: string) => void
+}
+
+// To generate Unique key for each Alert Component
+let seeds = 0
+const generateKey = (): string => {
+  return new Date().getTime() + (seeds++).toString()
+}
+
+//Transition/Animation Using react-spring useTransition
+const TransitionWrapper = (props: {
+  alerts: Array<AlertProps>
+  placement: placementType
+}) => {
+  const config: any = (item, state) => {
+    return {
+      duration: 250,
+      easing: t => {
+        if (state === "leave") {
+          return (t * t + t * t * t) / 2 // - 3 * t - 3 * t * t * t //Bezier Curve (1,0,1,1)
+        } else {
+          return t * t * t //Bezier Curve (0,0,0,1)
+        }
+      },
+    }
+  }
+  const [alerts, setAlerts] = useState(props.alerts)
+  const [refMap] = useState(() => new WeakMap())
```

Comment:
refs is the key part in the animation, because they are capturing the DOM node at the time of render and able to pass it's height property to react-spring transition config. Also it's weakMap so remove key-pair itself when deleted from any alert remove from state, as alerts are keys of refMap.